src/Shared/CopyOnWriteDictionary.cs

```diff
@@ -34,31 +30,20 @@ namespace Microsoft.Build.Collections
     /// be run in a separate appdomain.
     /// </comment>
     [Serializable]
-    internal class CopyOnWriteDictionary<K, V> : IDictionary<K, V>, IDictionary
+    internal class CopyOnWriteDictionary<K, V> : IDictionary<K, V>, IDictionary, ISerializable
```

Comment:
I didn't notice any places where this significantly diverged from ImmutableDictionary other than with the serialization part. Is there a problem with extending ImmutableDictionary and not overriding most of its methods?

Reply:
`ImmutableDictionary` is sealed.